src/main/java/nl/tudelft/lifetiles/annotation/model/ResistanceAnnotation.java

```diff
@@ -114,9 +112,34 @@ public String getGeneName() {
      * @return segment which annotation should be mapped to.
      */
     @Override
-    public SequenceSegment mapOntoSequence(
-            final List<SequenceSegment> segments, final Sequence reference) {
+    public SequenceSegment mapOntoSequence(final Set<SequenceSegment> segments,
+            final Sequence reference) {
+        for (SequenceSegment segment : segments) {
+            if (segment.getSources().contains(reference)
+                    && segment.getStart() <= getGenomePosition()
+                    && segment.getEnd() > getGenomePosition()) {
+                return segment;
+            }
+        }
         return null;
     }
 
+    /**
+     * {@inheritDoc}
```

Comment:
inherritDoc for toString is not the best idea, since you override to string to get away from the default implementation, and then the docs no longer describe the format the string will return